concept/type_/type_reader.rs

```diff
@@ -37,8 +37,7 @@ pub struct TypeReader { }
 impl<'_s> TypeReader
     where '_s : 'static {
```

Comment:
I still don't understand why we need `'_s: 'static`. `'static` is the only lifetime that outlives `'static`.

Reply:
Ah good catch. 
That's me being non-committal to requiring the parameters all be static. I'll revert.